File: bak_1931/sol.py
import sys
N = int(sys.stdin.readline())
time = [[0] * 2 for _ in range(N)]
for i in range(N):
    s, e = map(int, sys.stdin.readline().split())
    time[i][0] = s
    time[i][1] = e

time.sort(key = lambda x : (x[1], x[0]))

cnt = 1
end_time = time[0][1]
for i in range(1,N):
    if time[i][0] >= end_time:
        cnt += 1
        end_time = time[i][1]

print(cnt)

# 회의실 문제 출처: 끝나는 시간 오름차순 이랑 시작 시간 오름차순 
# 출처: https://suri78.tistory.com/26

File: bak_1931/sol.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

struct Time {
    int begin; // 회의 시작
    int end; // 회의 끝 
};

bool cmp(Time f, Time s){
    if (f.end == s.end)
        return f.begin < s.begin; // 종료 시간이 같다면 ,시작 시간이 빠른 순
    else
        return f.end < s.end;
}

int main(){
    int N; 
    cin >> N;

    vector<Time> t(N);
    for (int i = 0; i < N; i++){
        cin >> t[i].begin >> t[i].end;
    }

    sort(t.begin(), t.end(), cmp);

    int cnt = 0; // 회의가 가능한 수
    int n = 0; // 회의 종료 지점 저장 

    for(int i = 0; i < t.size(); i++){
        if (n <= t[i].begin) { // 회의 종료 지점이 다음 회의 시작지점보다 작으면
            n = t[i].end; // n 에 다음 회의 종료 지점 저장 
            cnt++; // 회의 가능 수 증가
            
        }

    }
    cout << cnt << endl;
}

// 출처 : https://kim6394.tistory.com/67